charts/kube-prometheus-stack/templates/prometheus/secret.yaml

```
@@ -10,6 +10,8 @@ metadata:
     app.kubernetes.io/component: prometheus
 {{ include "kube-prometheus-stack.labels" . | indent 4 }}
 data:
-  object-storage-configs.yaml: {{ toYaml .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret | b64enc | quote }}
+  object-storage-configs.yaml: {{- if .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secretIsBase64 }} {{ toYaml .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret | quote }}
+{{- else }} {{ toYaml .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret | b64enc | quote }}
+{{- end }}
```

I'm not really happy with that approach. What about

```suggestion
object-storage-configs.yaml: {{ tpl (toYaml .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret) $ | b64enc | quote }}
```

Then, you could do '{{ printf "%s" "base64str" | b64dec }}' on your values.

---

I think the `tpl (toYaml .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret) $` is a valid and a more generic approach which may cover more use-cases.